### ISSUE TITLE
Release files

### ISSUE THREAD

**Issue by nicostubi** [xmlunit/xmlunit#194]
Hello,

I am totally new to this package.

I have been able to successfully compare two XML files based on the following code:

`Diff myDiff = DiffBuilder.Compare(Input.FromFile(expected)).WithTest(Input.FromFile(result)).Build();
            
            // assert 
            Assert.False(myDiff.HasDifferences(), myDiff.ToString());`

In the Dispose() method of my test class, I want to get rid of the tested file.

I get the folllowing error:

The process cannot access the file 'C:\Projects\....xml' because it is being used by another process.

I see that DiffBuilder is not disposable. 

BR
Nicolas

**Comment by bodewig** [comment]
You are correct, at least in the case where XMLUnit opens the streams itself, there needs to be a way to close and dispose those streams. I'm just not sure where to put it most conveniently. `InputBuilder.FromFile` could return a disposable builder, probably, I don't really see a better place right now. Some of the `IStream` implementations, in particular `StreamSource` should probably be disposable as well, but you don't get access to them via the fluent API.

As a workaround for your case you could set up `FileStream`s for your files and use `Input.FromStream` which would allow you to close and dispose the streams yourself.

**Comment by bodewig** [comment]
actually, this should have been in https://github.com/xmlunit/xmlunit.net/ :-)

What I said is still true here, the builder returned by `Input.fromFile` should be `Closable` - `AutoClosable` if we didn't want to support Java 6. Java's built-in `StreamSource` is not closeable.

**Comment by nicostubi** [comment]
Do you want me to create the issue in https://github.com/xmlunit/xmlunit.net/ ?

Yes, I was looking for a disposable builder.

**Comment by bodewig** [comment]
> Do you want me to create the issue in https://github.com/xmlunit/xmlunit.net/ ?

No, that's not necessary, I'll need to handle the issue for Java and .NET anyway. Thanks.
